app/auth/defra-id/validate-initialisation-vector.js

```diff
@@ -0,0 +1,15 @@
+const { INITIALISATION_VECTOR } = require('../../constants/cache-keys')
+const { parseJwt } = require('./parse-jwt')
+
+const validateInitialisationVector = (request, accessToken) => {
+  const cachedInitialisationVector = request.yar.get(INITIALISATION_VECTOR)
+  const initialisationVector = parseJwt(accessToken).nonce
```

Comment:
The token we get back needs to be parsed to JSON so we can access the `nonce` property which containers the Initialisation Vector.

We can then compare it to the value we've already cached.